src/hooks/useEntitlement.ts

```diff
@@ -0,0 +1,51 @@
+import { useQueries } from 'react-query';
+import { useMemo } from 'react';
+
+import type { GetEntitlementsResponse } from '../../types/checkout';
+import type { MediaOffer } from '../../types/media';
+
+import { useConfigStore } from '#src/stores/ConfigStore';
+import { useAccountStore } from '#src/stores/AccountStore';
+import { getEntitlements } from '#src/services/checkout.service';
+
+export type UseEntitlementResult = {
+  isEntitled: boolean;
+  isLoading: boolean;
+  error: unknown | null;
+};
+
+export type UseEntitlement = (mediaOffers?: MediaOffer[], enabled?: boolean) => UseEntitlementResult;
+
+type QueryResult = {
+  responseData?: GetEntitlementsResponse;
+};
+
+const useEntitlement: UseEntitlement = (mediaOffers = [], enabled = true) => {
+  const sandbox = useConfigStore(({ config }) => config?.cleengSandbox);
+  const jwt = useAccountStore(({ auth }) => auth?.jwt);
+
+  const entitlementQueries = useQueries(
+    mediaOffers.map(({ offerId }) => ({
+      queryKey: ['mediaOffer', offerId],
+      queryFn: () => getEntitlements({ offerId: offerId || '' }, sandbox, jwt || ''),
```

Comment:
Should we just skip blank offer ID's instead of sending empty string?

Reply:
It is skipped due to the `enabled` property being set to `false`, we are doing this on multiple occasions but due to your comment, I guess this isn't clear.

Reply:
In this case, I think that we can ensure that all `mediaOffers` have a valid `offerId`. Then we can prevent skipping these queries entirely. According to the `MediaOffer` type, the `offerId` can only be a string. 

So technically, the or operator is redundant since it will be an empty string already. 

But I think removing invalid `offerIds` from the array is more clear.

```tsx
const useEntitlement: UseEntitlement = (mediaOffers = [], enabled = true) => {
  const sandbox = useConfigStore(({ config }) => config?.cleengSandbox);
  const jwt = useAccountStore(({ auth }) => auth?.jwt);

  // memorize since `.filter` creates a new Array 
  const validMediaOffers = useMemo(() => mediaOffers.filter(({ offerId }) => !!offerId), [mediaOffers]);

  const entitlementQueries = useQueries(
    validMediaOffers.map(({ offerId }) => ({
      queryKey: ['mediaOffer', offerId],
      queryFn: () => getEntitlements({ offerId }, sandbox, jwt || ''),
      enabled: enabled && !!jwt,
    })),
  );
```

Reply:
This will change if I take out `useQueries`. I hope I don't need the extra filtering, to be honest, but I'll keep your request in mind to prevent the empty string assignment.